src/App.tsx

```diff
@@ -1,26 +1,367 @@
 /* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { addTodo, deleteTodo, getTodos, updateTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Errors, Filters } from './types/enums';
+import { USER_ID } from './consts/consts';
+import { Footer } from './components/Footer/Footer';
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotification';
 
 export const App: React.FC = () => {
+  const [prepariedTodos, setPrepariedTodos] = useState<Todo[]>([]);
+
+  const [filterMethod, setFilterMethod] = useState(Filters.all);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const [todoTitle, setTodoTitle] = useState('');
+  const titleRef = useRef<HTMLInputElement>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [todoOnDeleting, setTodoOnDeleting] = useState<number | null>(null);
+  const [todosOnDeleting, setTodosOnDeleting] = useState<number[]>([]);
+
+  const [todoOnUpdating, setTodoOnUpdating] = useState<number | null>(null);
+  const [todosOnUpdating, setTodosOnUpdating] = useState<number[]>([]);
```

Comment:
we should simplify this states - they are the same
can we use just one state for add delete and update?